### PR TITLE
8 validate username password

### PR DESCRIPTION
**Pull request by RaeesaParker**
-> Add validation.js file in middleware 
-> Add function to iterate through wanted checks 
-> Add function to check password (followed the rule of 8 char, 1 number, 1 letter supplied by Demi in Figma)
-> Add function to check username is unique

### CODE REVIEW

**Comment by RaeesaParker** [comment]
Readded the "require dotenv" in the db file. I keep getting an error of "Dialect needs to be explicitly supplied as of v4.0.0" without this line. 

-> I've tested the user creation in Postman and everything seems to work. 

-> But I'm not sure how to reflect the changes on the swagger documentation

**Comment by mz106** [comment]
1. Cool, no worries. 
2. Nice
3. I'll check tomorrow

File: src/middleware/validation.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const User = require("../user/model.js");
```
Comment (mz106) [inline src/middleware/validation.js:1]: If you want to stretch yourself with the validation, then try it with triggers - 

const User = sequelize.define('user', {
  username: {
    type: Sequelize.STRING,
    allowNull: false
  },
  password: {
    type: Sequelize.STRING,
    allowNull: false
  }
}, {
  hooks: {
    beforeCreate: (user, options) => {
      const salt = bcrypt.genSaltSync();
      user.password = bcrypt.hashSync(user.password, salt);
    }
  }
});

It's up to you

**Review by mz106 — APPROVED** [review]
Approved. See my comment about triggers

**Pull request merged**